### PR TITLE
issue2167

### PR DESCRIPTION
**Pull request by Asnelchristian**
please have a look at this

### CODE REVIEW

File: coalib/collecting/Collectors.py [hunk 1]
```diff
@@ -214,7 +214,7 @@ def collect_all_bears_from_sections(sections, log_printer):
     """
     Collect all kinds of bears from bear directories given in the sections.
 
-    :param bear_dirs:   directory name or list of such that can contain bears
+    :param sections:   directory name or list of such that can contain bears
```
Comment (AbdealiJK) [inline coalib/collecting/Collectors.py]: The indentation here is wrong now, notice how the sentences begin at the same column for all the documentations ?

Like the "tuple" and "log_printer" in the lines below

File: coalib/collecting/Collectors.py [hunk 2]
```diff
@@ -214,7 +214,7 @@ def collect_all_bears_from_sections(sections, log_printer):
     """
     Collect all kinds of bears from bear directories given in the sections.
 
-    :param sections:   directory name or list of such that can contain bears
+    :param sections:    directory name or list of such that can contain bears
```
Comment (AbdealiJK) [inline coalib/collecting/Collectors.py]: Ah, you fix it here. Please squash both your commits into 1 commit

**Comment by AbdealiJK** [comment]
Hey there,

The git commit message you've used is not in accordance with the coala git commit guidelines.
Can you check out the [commit style documentation](http://coala.readthedocs.org/en/latest/Getting_Involved/Writing_Good_Commits.html) and ensure you follow the guidelines mentioned there ?

**Comment by sils1297** [comment]
Hi,

it looks like your PR needs squashing. [This page](http://coala.readthedocs.org/en/latest/Users/Tutorials/Git_Help.html) may help you to get started on this.

```
git rebase --interactive upstream/master
```

Now you can squash the commits together. You should also edit your commit message to comply to our guidelines: http://coala.readthedocs.io/en/latest/Getting_Involved/Writing_Good_Commits.html

Now push it up. You will need to do a force push because you change history here (do not do this on master branches :)):

```
git push --force
```

I know this is a lot and it's hard for every newcomer, you're not alone with that. Ping us on [our gitter channel](https://gitter.im/coala-analyzer/coala) if you need help!

File: coalib/collecting/Collectors.py [hunk 3]
```diff
@@ -214,7 +214,7 @@ def collect_all_bears_from_sections(sections, log_printer):
     """
     Collect all kinds of bears from bear directories given in the sections.
 
-    :param bear_dirs:   directory name or list of such that can contain bears
+    :param sections:    directory name or list of such that can contain bears
```
Comment (sils1297) [inline coalib/collecting/Collectors.py]: I think the comment needs adjustment as well

**Comment by sils1297** [comment]
Commit message shortlog `Collectors.py: change bear_dirs -> sections` could maybe be rather `Collectors.py: Co`

**Comment by sils1297** [comment]
`

**Comment by sils1297** [comment]
`Collectors: Correct section param docs`

(Note the upper case C after the colon)

**Comment by sils1297** [comment]
https://github.com/coala-analyzer/coala/pull/2173/commits/73b0259bafe082ef20eaf7d2588122900e8cba59 needs work

File: coalib/collecting/Collectors.py (hunk 3 above)
Comment (sils1297) [inline coalib/collecting/Collectors.py]: what I'm saying here is that `sections` is no `directory name or list of such that can contain bears`!

File: coalib/collecting/Collectors.py [hunk 4]
```diff
@@ -150,11 +150,11 @@ def collect_bears(bear_dirs, bear_globs, kinds, log_printer,
     Collect all bears from bear directories that have a matching kind
     matching the given globs.
 
-    :param bear_dirs:           Directory name or list of such that can contain
+    :param bear_dirs:           directory name or list of such that can contain
```
Comment (Adrianzatreanu) [inline coalib/collecting/Collectors.py]: first of all, why are we uncapitalizing them? I think they should stay capitalized, why would we change them?

**Comment by Adrianzatreanu** [comment]
c1fe797 needs work imo

**Comment by Adrianzatreanu** [comment]
unack c1fe797

**Comment by Asnelchristian** [comment]
what is wrong ?

File: coalib/collecting/Collectors.py [hunk 5]
```diff
@@ -154,7 +154,7 @@ def collect_bears(bear_dirs, bear_globs, kinds, log_printer,
                                 bears.
     :param bear_globs:          Globs of bears to collect.
     :param kinds:               List of bear kinds to be collected.
-    :param log_printer:         LogPrinter to handle logging.
+    :param log_printer:         log_printer to handle logging.
```
Comment (Adrianzatreanu) [inline coalib/collecting/Collectors.py]: hey I don't understand why this is needed.

Comment (Adrianzatreanu) [inline coalib/collecting/Collectors.py]: ``log_printer to handle logging`` this is about ``from coalib.output.printers.LogPrinter import LogPrinter`` this, so I think it was okay the way it was.

File: coalib/collecting/Collectors.py [hunk 6]
```diff
@@ -214,7 +214,7 @@ def collect_all_bears_from_sections(sections, log_printer):
     """
     Collect all kinds of bears from bear directories given in the sections.
 
-    :param bear_dirs:   directory name or list of such that can contain bears
+    :param sections:    list of sections from which bears are to be collected
```
Comment (Adrianzatreanu) [inline coalib/collecting/Collectors.py]: run ``coala`` over your code to fix inconsistencies in code

File: coalib/collecting/Collectors.py (hunk 5 above)
Comment (Asnelchristian) [inline coalib/collecting/Collectors.py]: there are other functions in the same file having  log_printer as parameter;  
they all have the same param docs for log_printer that is why i modified LogPrinter to have something uniform

Comment (Adrianzatreanu) [inline coalib/collecting/Collectors.py]: hm it seems you are right.

**Comment by Adrianzatreanu** [comment]
seems your commit ``head`` also has 54 chars, whereas 50 maximum allowed: make it shorter

File: coalib/collecting/Collectors.py (hunk 6 above)
Comment (Asnelchristian) [inline coalib/collecting/Collectors.py]: i ran coala, why is this still here?

**Comment by sils1297** [comment]
Hey, your code changes look good, can you squash your commits again though so we have a clean history?

I know it might seem very cumbersome but you'll get accustomed to this rather fast and once you've learned it it'll be way easier for you to do higher quality work :)

**Comment by Asnelchristian** [comment]
i am sorry for all the trouble, i directly try to use 'git rebase -i   ' but i made some mistake 
How can i arrange this

**Comment by sils1297** [comment]
Hey, GitMate has commented on the "outdated diff" as github calls it. What that means is that something's wrong with your first commit that you correct in your second - however we'd prefer to have it right in the first commit right ahead :) Can you do the interactive rebase again, edit the first commit and run coala there? Then do `git commit -a --amend` to change the commit and continue the rebase.

Feel free to ask around if you need help, you know that I guess ;)

**Comment by sils1297** [comment]
also please rebase while you're doing this, you know that game by now I guess?

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2173/commits/c28463152cf44ac427e868e21f944e49708c5ec8

can you rebase it another time? It's out of date again. Ping us when you push so we can merge it before other stuff gets merged. The appveyor problem seems to be unrelated to your commit so it's our problem not yours :)

**Comment by sils1297** [comment]
Hey, that didn't seem to work, maybe ask on gitter for assistance?

Add the original coala repository as a remote:

```
git remote add upstream https://github.com/coala-analyzer/coala.git
```

Then go to your branch and rebase it onto our master:

```
git checkout <yourbranch>
git fetch upstream
git rebase --interactive upstream/master
```

Now push it up. You will need to do a force push because you change history here (do not do this on master branches :)):

```
git push --force
```

**Comment by sils1297** [comment]
reack https://github.com/coala-analyzer/coala/pull/2173/commits/81ea8bcf0dddbaefb45f1f8cf2bf5acb82e11382

**Comment by arafsheikh** [comment]
@rultor merge

**Comment by arafsheikh** [comment]
@rultor merge

**Comment by arafsheikh** [comment]
@rultor merge

**Pull request merged**